addons/mail/static/src/owl/components/mail_template/mail_template.js

```diff
@@ -0,0 +1,62 @@
+odoo.define('mail.component.MailTemplate', function (require) {
+'use strict';
+
+const useStore = require('mail.hooks.useStore');
+
+const { Component } = owl;
+const { useDispatch } = owl.hooks;
+
+
+class MailTemplate extends Component {
+
+    /**
+     * @override
+     * @param {...any} args
+     */
+    constructor(...args) {
+        super(...args);
+        this.storeDispatch = useDispatch();
+        this.storeProps = useStore((state, props) => {
+            const mailTemplate = state.mailTemplates[props.mailTemplateLocalId];
+            const activity = state.activities[props.activityLocalId];
+            return { activity, mailTemplate };
+        });
+    }
+
+    //--------------------------------------------------------------------------me
+    // Handlers
+    //--------------------------------------------------------------------------
+
+    /**
+     * @private
+     * @param {MouseEvent} ev
+     */
+    _onClickPreview(ev) {
+        ev.stopPropagation();
+        ev.preventDefault();
+        this.storeDispatch('previewMailTemplate', this.props.mailTemplateLocalId, this.props.activityLocalId);
+    }
+
+    /**
+     * @private
+     * @param {MouseEvent} ev
+     */
+    async _onClickSend(ev) {
+        ev.stopPropagation();
+        ev.preventDefault();
+        this.storeDispatch('sendMailTemplate', this.props.mailTemplateLocalId, this.props.activityLocalId);
+
+    }
+}
+
+Object.assign(MailTemplate, {
+    props: {
+        activityLocalId: String,
```

Comment:
I feel like `activityLocalId` should be optional for this component

Reply:
the preview and the send requires an activity to be defined as it is used as context for the mail so without activity the mail template used here does not really have sense, don't you think ?

Reply:
Haven't taken a look at messaging actions yet, but I had the impression that component `MailTemplate` acts a bit like component `Message`, in that `Message` component needs a `message` but `thread` is optional, which I thought could be similar with `MailTemplate` and its use of `activity`.

Reply:
Maybe but then we need to hide Preview and send button ? As the activity data are required to make the rpc calls needed in these button actions ?

Reply:
(I lost you ?)